astroid/raw_building.py

```diff
@@ -186,14 +186,27 @@ def object_build_class(node, member, localname):
     basenames = [base.__name__ for base in member.__bases__]
     return _base_class_object_build(node, member, basenames, localname=localname)
 
+def _get_args_info_from_callable(member: Callable) -> Tuple[List[str], List[str], List[str], List[str]]:
+    """
+    Returns args, posonlyargs, defaults, kwonlyargs. 
 
-def object_build_function(node, member, localname):
-    """create astroid for a living function object"""
-    signature = inspect.signature(member)
+    :note: currently ignores the return annotation.
+    """
     args = []
     defaults = []
     posonlyargs = []
     kwonlyargs = []
+
+    try:
+        signature = inspect.signature(member)
+    except ValueError:
+        # function either has an invalid text signature or no signature
+        # at all. We distinguish between the two by looking at the
+        # __text_signature__ attribute
+        if getattr(member, "__text_signature__", None) is not None:
+            f"Cannot parse __text_signature__ signature of {member}"
```

Comment:
Can we log a warning from astroid?

Reply:
We could but do not, generally we raise ``InferenceError`` see this issue in flake8 where a log take 6% of the exec time even when not displayed : https://github.com/PyCQA/flake8/pull/1544

Reply:
Should we raise `InferenceError` here though? 
It's not technically inference what we're doing here.

Reply:
Yeah, setting back to `None` seems the best option here. That mimics what we currently do.